modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -188,6 +201,21 @@ public UserAccount putAssetLibraryUserAccount(
 			assetLibraryId, _updateUser(assetLibraryId, userAccountId, true));
 	}
 
+	private void _checkAssetLibraryUser(long assetLibraryId) throws Exception {
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		if (permissionChecker.isGroupAdmin(assetLibraryId)) {
+			return;
+		}
+
+		if (!_userService.hasGroupUser(
+				assetLibraryId, contextUser.getUserId())) {
+
+			throw new NotFoundException();
```

Comment:
Here you're throwing a different exception (NoSuchUser* vs NotFound*). Shouldn't you throw the same error in both cases?

Reply:
I think there are 2 different cases:

* NotFoundException is returned when the current user is not member of the Space. It has no rights to call the asset library endpoint.
* NotSuchUserException is returned when the current user is a member of the Space. We can give him the info that a user doesn't belong to the Space.

Reply:
I've replaced the `NotFoundException` with `PrincipalException.MustHavePermission` which seems more suitable and it is what a permission check normally returns on error. However at the end we receive a NOT_FOUND status from the endpoint, not a 401. This is why I started by sending the `NotFoundException`, I was checking the return status of endpoints.